source/common/router/config_impl.cc

```diff
@@ -261,26 +283,23 @@ const Route* RouteEntryImplBase::clusterEntry(uint64_t random_value) const {
       // sum(weights) > WeightedClusterEntry::MAX_CLUSTER_WEIGHT.
       // In this case, terminate the search and just return the cluster
       // whose weight cased the overflow
-      return cluster.get();
+      return cluster;
     }
     begin = end;
   }
   NOT_IMPLEMENTED;
 }
 
 void RouteEntryImplBase::validateClusters(Upstream::ClusterManager& cm) const {
-  // Throws an error if any of the clusters held by this Route or the
-  // internal weighted clusters are invalid.
   if (isRedirect()) {
     return;
   }
 
-  // We could have either a normal cluster or weighted cluster
-  if (weighted_clusters_.empty()) {
-    if (!cm.get(this->clusterName())) {
-      throw EnvoyException(fmt::format("route: unknown cluster '{}'", this->clusterName()));
+  if (!cluster_name_.empty()) {
+    if (!cm.get(cluster_name_)) {
```

Comment:
Doesnt this result in error if in config I specify routes with clusters, that do not yet exist (because the cds query has not yet been made)?